microsoft/testsuites/cvm/cvm_boot.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from pathlib import Path
+from typing import Any, List, cast
+
+from assertpy.assertpy import assert_that, assert_warn
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+)
+from lisa.features.security_profile import (
+    CvmEnabled,
+    SecurityProfile,
+    SecurityProfileSettings,
+)
+from lisa.operating_system import CBLMariner, Posix
+from lisa.sut_orchestrator import AZURE
+from lisa.testsuite import TestResult, simple_requirement
+from lisa.tools import Lsblk, Reboot, Tpm2
+from lisa.tools.lsblk import PartitionInfo
+from lisa.util import (
+    SkippedException,
+    TcpConnectionException,
+    UnsupportedDistroException,
+    constants,
+)
+from lisa.util.shell import wait_tcp_port_ready
+
+
+@TestSuiteMetadata(
+    area="cvm",
+    category="functional",
+    description="""This test suite covers some common scenarios related to
+    CVM boot on Azure.
+    """,
+)
+class CVMBootTestSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node: Node = kwargs["node"]
+        if not isinstance(node.os, CBLMariner):
+            raise SkippedException(
+                UnsupportedDistroException(
+                    node.os, "CVM boot test supports only Azure Linux."
+                )
+            )
+
+    @TestCaseMetadata(
+        description="""This test verifies that TPM enrollment is done correctly on
+        a CVM with encrypted root partition
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_features=[CvmEnabled()],
+            supported_platform_type=[AZURE],
+        ),
+    )
+    def verify_encrypted_root_partition(
+        self,
+        log: Logger,
+        node: RemoteNode,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+    ) -> None:
+        security_profile_settings = cast(
+            SecurityProfileSettings, node.features[SecurityProfile].get_settings()
+        )
+        if not security_profile_settings.encrypt_disk:
+            raise SkippedException("This test requires disk encryption to be enabled")
+        lsblk = node.tools[Lsblk]
+        disks = lsblk.get_disks(force_run=True)
+        partitions: List[PartitionInfo] = next(
+            (d.partitions for d in disks if d.name == "sda"), []
+        )
```

Comment:
please use below code to get root partition, since fixed label name may be changed when VM reboot
node_disk = node.features[Disk]
node_disk.get_os_boot_partition()

Reply:
I want to get the root partition rather than boot partition, but following the suggestion, I added a call to `bootctl` to get the root device path more reliably.